addons/mrp/models/product_document.py

```diff
@@ -0,0 +1,24 @@
+
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class ProductDocument(models.Model):
+    _inherit = 'product.document'
+
+    def _default_attached_on_mrp(self):
+        return "bom" if self.env.context.get('attached_on_bom') else "hidden"
+
+    attached_on_mrp = fields.Selection(
```

Comment:
I don't understand why you don't reuse `attached_on`? A single document can be on a BoM *and* on a quotation?

Reply:
The document indeed could be "linked" to multiple objects.